lib/net.js

```diff
@@ -894,6 +906,12 @@ Socket.prototype.connect = function(options, cb) {
   self._connecting = true;
   self.writable = true;
 
+  // Cork then uncork upon connecting.
+  this.cork();
+  this.once('connect', function() {
+    this.uncork();
+  });
```

Comment:
It is supposed to keep the initial write (if written to multiple times when connecting) inside the stream buffer - then, when it connects (and is writable), all the writes can be flushed at once (with writev), rather than a single write then a writev with the rest.